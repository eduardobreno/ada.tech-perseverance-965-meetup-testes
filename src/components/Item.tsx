import React from 'react';
import {Text, StyleSheet, View, Switch} from 'react-native';

import Button from './Button';

export interface ItemType {
  name: string;
  done: boolean;
}

interface ItemProps {
  item: ItemType;
  toggle: (item: ItemType) => void;
  remove: (item: ItemType) => void;
}

const Item = ({item, toggle, remove}: ItemProps) => (
  <View key={item.name} style={styles.item}>
    <Switch
      testID={`toggle${item.name}`}
      style={styles.toggle}
      onValueChange={() => toggle(item)}
      value={item.done}
    />

    <Text style={styles.text} numberOfLines={2} ellipsizeMode="tail">
      {item.name}
    </Text>
    <Button
      testID={`remove${item.name}`}
      title="x"
      onPress={() => remove(item)}
    />
  </View>
);

const styles = StyleSheet.create({
  item: {
    paddingBottom: 15,
    flexDirection: 'row',
    alignItems: 'center',
  },
  toggle: {
    marginRight: 10,
  },
  text: {
    flex: 1,
    flexWrap: 'wrap',
  },
});

export default Item;
