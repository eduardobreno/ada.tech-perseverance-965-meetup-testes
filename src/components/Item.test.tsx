import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import Item, {ItemType} from './Item';

it('should render Item component', async () => {
  const item: ItemType = {name: 'Item 01', done: false};
  const remove = jest.fn();
  const toggle = jest.fn();

  render(<Item item={item} remove={remove} toggle={toggle} />);

  fireEvent(screen.getByTestId(`toggle${item.name}`), 'valueChange', true);
  fireEvent.press(screen.getByText('x'));

  expect(remove).toBeCalledWith(item);
  expect(toggle).toBeCalledWith(item);
  expect(screen.toJSON()).toMatchSnapshot();
});
