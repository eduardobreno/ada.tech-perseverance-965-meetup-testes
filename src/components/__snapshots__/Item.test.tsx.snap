// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Item component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingBottom": 15,
    }
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "height": 31,
          "width": 51,
        },
        {
          "marginRight": 10,
        },
      ]
    }
    testID="toggleItem 01"
    value={false}
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={2}
    style={
      {
        "flex": 1,
        "flexWrap": "wrap",
      }
    }
  >
    Item 01
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#000000",
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 25,
        "paddingVertical": 10,
      }
    }
    testID="removeItem 01"
  >
    <Text>
      x
    </Text>
  </View>
</View>
`;
